cmd/root.go

```diff
@@ -709,8 +753,16 @@ func run(cmd *Command) error {
 	}
 
 	if !cmd.cfg.DisableReload {
-		// start watching for file changes to trigger dynamic reloading
-		go watchFile(ctx, cmd.tools_file, s)
+
+		// extract parent dir for relevant files and dedup
+		for _, f := range relevantFiles {
+			cleanFile := filepath.Clean(f)
+			watchedFiles[cleanFile] = true
+			watchDirs[filepath.Dir(cleanFile)] = true
+		}
```

Comment:
nit: currently there is a vary long gap between initialization here and usage (especially for watchedFiles and watchDirs). Can we move things closer together?

One reasonable thing might be to move this into a function that returns those things and then call `go watchcChanges` inside section above.